### PR TITLE
build: Add placeholder packages for npm binary distributions

### PR DESCRIPTION
**Pull request by lforst**
Adds packages for binary distributions on npm based on the binaries we currently distribute.

Ref https://github.com/getsentry/sentry-cli/issues/1360#issuecomment-1819270364

### CODE REVIEW

**Review by szokeasaurusrex — APPROVED** [review]
Looks good (I am trusting that you have checked that the `package.json` are correct; since I am not super familiar with how these work)

File: npm-binary-distributions/darwin/package.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "@sentry/cli-darwin",
```
Comment (szokeasaurusrex) [inline npm-binary-distributions/darwin/package.json:2]: Do we need to have a separate binary for M1/M2 and Intel Macs?

Comment (lforst) [reply npm-binary-distributions/darwin/package.json:2]: The CLI is currently set up to use our universal darwin binary no matter what architecture.

https://github.com/getsentry/sentry-cli/blob/1c5d20f94043ac66631157b2a208d5668541921c/scripts/install.js#L78-L80

Also, I believe mac OS has a compatibility layer for non-arm binaries. But good that you confirmed!

Comment (szokeasaurusrex) [reply npm-binary-distributions/darwin/package.json:2]: Yeah, but I believe the Rosetta compatibility layer you are referencing imposes a performance penalty, and also, if it is the first time the user is running a non-arm binary, the user is prompted to install Rosetta and needs to wait some time for the install to complete. These are probably only minor inconveniences, but we should probably be creating a native ARM build (as we already are with the universal binary)

**Pull request merged**